packages/frontend/src/hooks/useTxResult.ts

```diff
@@ -0,0 +1,46 @@
+import { useQuery } from 'react-query'
+import { Token } from '@hop-protocol/sdk'
+import Network from 'src/models/Network'
+import { BigNumber } from 'ethers'
+
+export function useTxResult(
+  token?: Token,
+  srcNetwork?: Network,
+  destNetwork?: Network,
+  amount?: BigNumber,
+  fn?: (opts: any) => any,
+  opts?: any
+) {
+  const queryKey = `txResult:${srcNetwork?.slug}:${destNetwork?.slug}:${amount?.toString()}`
+  const { interval, ...rest } = opts
+
+  const { isLoading, isError, data, error } = useQuery(
+    [queryKey, srcNetwork?.slug, token?.symbol, amount?.toString()],
+    async () => {
+      if (!(token && srcNetwork?.slug && destNetwork?.slug && amount && fn)) {
+        return
+      }
+
+      const options = {
+        token: token,
+        fromNetwork: srcNetwork,
```

Comment:
minor suggestion:

This suggestion applies to the entire application;
it would be good to choose consistent naming for the network directions; `fromNetwork` vs `sourceNetwork`

The contracts refer to them as `sourceChain` and `destinationChain` so maybe we use that convention throughout

Reply:
yeah, i was doing some thinking about `Network` / `Chain` models as well, and i believe we could consolidate them (just use `Chain`). totally agree about keeping consistent naming. for now, changing `fromNetwork` -> `sourceChain` and `toNetwork` -> `destinationChain` causes a huge diff, so i will address this in a more focused, near-future commit/pr.